Makefile

```diff
@@ -3,44 +3,68 @@ TIDY := $(shell which clang-tidy clang-tidy-8 2> /dev/null | tail -1 | xargs bas
 VALGRIND := valgrind
 
 SRC_DIR := src
+TEST_DIR := test
 OUT_DIR := out
+OUT_TEST_DIR := $(OUT_DIR)/test
 INCLUDE_TOP_DIR := include
 
 OUT_BIN_NAME := DG
 OUT_BIN := $(OUT_DIR)/$(OUT_BIN_NAME)
+OUT_TEST_BIN_NAME := DG_test
+OUT_TEST_BIN := $(OUT_TEST_DIR)/$(OUT_TEST_BIN_NAME)
+
+SRC_FILES := $(shell find $(SRC_DIR)/ -name *.cxx)
+TEST_SRC_FILES := $(shell find $(TEST_DIR)/ -name *.cxx)
+TEST_DEPS := $(filter-out $(SRC_DIR)/main.cxx,$(SRC_FILES)) $(TEST_SRC_FILES)
 
-SRC_FILES := $(wildcard */*.cxx)
 # Define OBJ_FILES in case wanting to use in future
 OBJ_FILES := $(SRC_FILES:.cxx=.o)
 VALGRIND_OUTPUT_FAILURE_TAG := LEAK
 
-CXX_FLAGS := -I$(INCLUDE_TOP_DIR)
+INCS := -I$(INCLUDE_TOP_DIR)
+TEST_INCS := -I/opt/gtest/include/
+
+# NOTE include flags __need__ to be on the left end of arguments.
```

Comment:
In test flags they are not. Remove or provide explanation why they need t obe on left end of arguments.

Reply:
Fix in: 472ac17